apps/etterlatte-saksbehandling-ui/client/src/components/behandling/aktivitetsplikt/aktivitetspliktTidslinje/AktivitetPeriode.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  AktivitetspliktType,
+  aktivitetspliktTypeTilLesbarStreng,
+  IAktivitetPeriode,
+  OpprettAktivitetPeriode,
+} from '~shared/types/Aktivitetsplikt'
+import { IDetaljertBehandling } from '~shared/types/IDetaljertBehandling'
+import {
+  AktivitetspliktRedigeringModus,
+  AktivitetspliktSkjemaAaVise,
+} from '~components/behandling/aktivitetsplikt/aktivitetspliktTidslinje/AktivitetspliktTidslinje'
+import { useForm } from 'react-hook-form'
+import { Button, HStack, Select, Textarea, VStack } from '@navikt/ds-react'
+import { ControlledDatoVelger } from '~shared/components/datoVelger/ControlledDatoVelger'
+import React from 'react'
+import { FloppydiskIcon, XMarkIcon } from '@navikt/aksel-icons'
+import { isPending } from '~shared/api/apiUtils'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { opprettAktivitetPeriodeForBehandling, opprettAktivitetPeriodeForSak } from '~shared/api/aktivitetsplikt'
+import { formaterTilISOString } from '~utils/formatering/dato'
+
+interface AktivitetPeriodeSkjema {
+  fom: string
+  tom?: string
+  type: AktivitetspliktType
+  beskrivelse: string
+}
+
+interface Props {
+  behandling?: IDetaljertBehandling
+  sakId: number
+  aktivitetPeriode?: IAktivitetPeriode
+  setAktivitetPerioder: (aktivitetPerioder: IAktivitetPeriode[]) => void
+  setAktivitetspliktRedigeringModus: (aktivitetspliktRedigeringModus: AktivitetspliktRedigeringModus) => void
+}
+
+export const AktivitetPeriode = ({
+  behandling,
+  sakId,
+  aktivitetPeriode,
+  setAktivitetPerioder,
+  setAktivitetspliktRedigeringModus,
+}: Props) => {
+  const [opprettAktivitetPeriodeForBehandlingResult, opprettAktivitetPeriodeForBehandlingRequest] = useApiCall(
+    opprettAktivitetPeriodeForBehandling
+  )
+  const [opprettAktivitetPeriodeForSakResult, opprettAktivitetPeriodeForSakRequest] =
+    useApiCall(opprettAktivitetPeriodeForSak)
+
+  const {
+    register,
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<AktivitetPeriodeSkjema>({
+    defaultValues: !!aktivitetPeriode
+      ? {
+          ...aktivitetPeriode,
+        }
+      : {
+          fom: '',
+          tom: '',
+          type: undefined,
+          beskrivelse: '',
+        },
+  })
+
+  const lagreAktivitetPeriode = (data: AktivitetPeriodeSkjema) => {
+    const aktivitetPeriodeRequestBody: OpprettAktivitetPeriode = !!aktivitetPeriode
+      ? {
+          fom: formaterTilISOString(data.fom),
+          tom: data.tom ? formaterTilISOString(data.tom) : '',
+          type: data.type,
+          beskrivelse: data.beskrivelse,
+          id: aktivitetPeriode.id,
+          sakId,
+        }
+      : {
+          fom: formaterTilISOString(data.fom),
+          tom: data.tom ? formaterTilISOString(data.tom) : '',
+          type: data.type,
+          beskrivelse: data.beskrivelse,
+          id: '',
```

Comment:
Ser du sender med tom streng for id en del steder -- fungerer det mot backend? Tror backend vil potensielt se på dette som id'en tom streng, i stedet for at det mangler en id. Fungerer det slik det er nå å lagre nye hendelser, med riktige id'er i backend?